### PR TITLE
Simple test and general improvements

### PR DESCRIPTION
**Pull request by markusressel**
Initially this PR was meant to be about tests, but I had to change the structure of the code a bit to be able to test it and got taken away... So in the end this is more or less a general improvements PR then :smile: 

~~In it's current form the `PIL` dependency is not needed anymore (I simplified the BytesIO code) so technically could be removed from the requirements. If there will be some kind of image text analysis in the future it might be needed again though.~~
I removed it.

I moved the initialization code to the `__main__` method so the bot doesn't start when you import the file :smile: 

I initially extracted the `BOT_TOKEN` constant to a `const.py` to remove it from the import phase as well. Later I moved the code to the `__main__` too so this technically is not necessary anymore but will probably come in handy once a proper config is in place (see #2 ). The config is currently just a dict but I would really like to have IDE support for the possible config values so this is only temporary.

I added lots of documentation too which hopefully reflects what the methods are really about.

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -17,87 +17,158 @@
...
     LOGGER.debug('Got image URL from the pool: {}'.format(url))
     return url
 
-def get_image():
-    url = get_image_url()
+
+def download_image_bytes(url: str) -> bytes:
+    """
+    Downloads the image from the given url
+    :return:
```
Comment (ekeih) [inline bot.py]:
```suggestion
    :return: the downloaded image
```

File: bot.py [hunk 2]
```diff
@@ -17,87 +17,158 @@
...
+    bot.send_message(chat_id=update.message.chat_id,
+                     text='Send /inspire for more inspiration :) Or use @InfiniteWisdomBot in a group chat and select one of the suggestions.')
 
-def send_random_quote(bot, update):
+
+def send_random_quote(bot: Bot, update: Update) -> None:
+    """
+    Sends a quote from the pool to the requesting chat
+    :param bot: the bot
+    :param update: the chat updater object
```
Comment (ekeih) [inline bot.py]:
```suggestion
    :param update: the chat update object
```

File: bot.py [hunk 3]
```diff
@@ -17,87 +17,158 @@
...
+    image_bytes_io = BytesIO(image_data)
+    image_bytes_io.name = 'inspireme.jpeg'
+    bot.send_photo(chat_id=chat_id, photo=image_bytes_io)
+
+
+def inlinequery(bot: Bot, update: Update) -> None:
+    """
+    Responds to an inline client request with a list of 16 randomly chosen images
+    :param bot: the bot
+    :param update: the chat updater object
```
Comment (ekeih) [inline bot.py]:
```suggestion
    :param update: the chat update object
```

Comment (markusressel) [reply bot.py]: done

File: bot.py (hunk 2 above)
Comment (markusressel) [reply bot.py]: done

File: bot.py (hunk 1 above)
Comment (markusressel) [reply bot.py]: done

**Pull request merged**